llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -2056,6 +2056,14 @@ void DwarfDebug::beginInstruction(const MachineInstr *MI) {
     }
   }
 
+  auto RecordSourceLine = [&](auto &DL, auto Flags) {
+    SmallString<128> LocationString;
+    raw_svector_ostream OS(LocationString);
+    DL.print(OS);
+
+    const MDNode *Scope = DL.getScope();
+    recordSourceLine(DL.getLine(), DL.getCol(), Scope, Flags, LocationString);
```

Comment:
Perhaps we should change `recordSourceLine()` to accept `DebugLoc` argument, instead of Line/Col/LocationString, ad do all the data extraction from `DL` there, in one place. That would obviate the need for this lambda.